internal/datastore/mysql/readwrite.go

```diff
@@ -0,0 +1,288 @@
+func exactRelationshipClause(r *v1.Relationship) sq.Eq {
+	return sq.Eq{
+		colNamespace:        r.Resource.ObjectType,
+		colObjectID:         r.Resource.ObjectId,
+		colRelation:         r.Relation,
+		colUsersetNamespace: r.Subject.Object.ObjectType,
+		colUsersetObjectID:  r.Subject.Object.ObjectId,
+		colUsersetRelation:  stringz.DefaultEmpty(r.Subject.OptionalRelation, datastore.Ellipsis),
+	}
+}
+
+func (rwt *mysqlReadWriteTXN) DeleteRelationships(filter *v1.RelationshipFilter) error {
+	// TODO (@vroldanbet) dupe from postgres datastore - need to refactor
+	ctx, span := tracer.Start(datastore.SeparateContextWithTracing(rwt.ctx), "DeleteRelationships")
+	defer span.End()
+
+	// Add clauses for the ResourceFilter
+	query := rwt.DeleteTupleQuery.Where(sq.Eq{colNamespace: filter.ResourceType})
+	tracerAttributes := []attribute.KeyValue{common.ObjNamespaceNameKey.String(filter.ResourceType)}
+	if filter.OptionalResourceId != "" {
+		query = query.Where(sq.Eq{colObjectID: filter.OptionalResourceId})
+		tracerAttributes = append(tracerAttributes, common.ObjIDKey.String(filter.OptionalResourceId))
+	}
+	if filter.OptionalRelation != "" {
+		query = query.Where(sq.Eq{colRelation: filter.OptionalRelation})
+		tracerAttributes = append(tracerAttributes, common.ObjRelationNameKey.String(filter.OptionalRelation))
+	}
+
+	// Add clauses for the SubjectFilter
+	if subjectFilter := filter.OptionalSubjectFilter; subjectFilter != nil {
+		query = query.Where(sq.Eq{colUsersetNamespace: subjectFilter.SubjectType})
+		tracerAttributes = append(tracerAttributes, common.SubNamespaceNameKey.String(subjectFilter.SubjectType))
+		if subjectFilter.OptionalSubjectId != "" {
+			query = query.Where(sq.Eq{colUsersetObjectID: subjectFilter.OptionalSubjectId})
+			tracerAttributes = append(tracerAttributes, common.SubObjectIDKey.String(subjectFilter.OptionalSubjectId))
+		}
+		if relationFilter := subjectFilter.OptionalRelation; relationFilter != nil {
+			query = query.Where(sq.Eq{colUsersetRelation: stringz.DefaultEmpty(relationFilter.Relation, datastore.Ellipsis)})
+			tracerAttributes = append(tracerAttributes, common.SubRelationNameKey.String(relationFilter.Relation))
+		}
+	}
+
+	span.SetAttributes(tracerAttributes...)
+
+	query = query.Set(colDeletedTxn, rwt.newTxnID)
+
+	querySQL, args, err := query.ToSql()
+	if err != nil {
+		return fmt.Errorf(errUnableToDeleteRelationships, err)
+	}
+
+	if _, err := rwt.tx.ExecContext(ctx, querySQL, args...); err != nil {
+		return fmt.Errorf(errUnableToDeleteRelationships, err)
+	}
+
+	return nil
+}
+
+func (rwt *mysqlReadWriteTXN) WriteNamespaces(newNamespaces ...*core.NamespaceDefinition) error {
+	// TODO (@vroldanbet) dupe from postgres datastore - need to refactor
+	ctx := datastore.SeparateContextWithTracing(rwt.ctx)
+
+	ctx, span := tracer.Start(ctx, "WriteNamespaces")
+	defer span.End()
+
+	// span.SetAttributes(common.ObjNamespaceNameKey.String(newNamespace.Name))
```

Comment:
Why is this commented out?

Reply:
Oh, because I didn't know what to do when there were more than one namespace. Suggestions?